### PR TITLE
Added ttl method

### PR DESCRIPTION
**Pull request by robinvdvleuten**
Hi there 👋 I've added a method to check for the end of the current duration. This way you will be able to add all standard rate-limit headers (X-RateLimit-Limit, X-RateLimit-Remaining, and X-RateLimit-Reset). It would be great if each Laravel cache implementation handles such functionality, but for now this is the best we can do. Hopefully you like it 😄

### CODE REVIEW

File: src/Throttlers/CacheThrottler.php [hunk 1]
```diff
@@ -147,6 +147,20 @@ public function count()
...
 
+    /**
+     * Get the time until the end of current duration.
+     *
+     * @return int
+     */
+    public function ttl()
+    {
+        if (!$this->store instanceof RedisStore) {
+            throw new \LogicException('The ttl() method can only be called for Redis backed throttlers');
```
Comment (GrahamCampbell) [inline src/Throttlers/CacheThrottler.php]: I don't think we should throw an exception here, since it means the code isn't driver agnostic anymore.

Comment (robinvdvleuten) [reply src/Throttlers/CacheThrottler.php]: So what would be better than to indicate the ttl() method is not supported? Maybe return a negative number?

**Comment by GrahamCampbell** [comment]
Thanks for the PR. I've left a comment.

**Comment by robinvdvleuten** [comment]
@GrahamCampbell I saw it and understand that throwing an exception is maybe not the best way to indicate that the ttl() method is not supported. We could mimic Redis's own ttl() return value though, which is a -1 to indicate that the key does not have an associated expire (https://redis.io/commands/ttl).

**Comment by GrahamCampbell** [comment]
Hmm. I'd probably recommend using Laravel's built in throttler if you need this feature?

**Comment by robinvdvleuten** [comment]
That's true I guess. But I thought it would make a nice addition to this library as well.

**Comment by the94air** [comment]
@robinvdvleuten  . Does that require the use of Redis?

**Comment by robinvdvleuten** [comment]
@the94air that's correct

**Comment by the94air** [comment]
😢 . I handled that long time ago by using the build in laravel login throttle trait and doing this:
```php
if (! function_exists('secondsToTime')) {
	function secondsToTime($seconds) {
	    $hours = floor($seconds / (60 * 60));
	 
	    $divisor_for_minutes = $seconds % (60 * 60);
	    $minutes = floor($divisor_for_minutes / 60);

	    $divisor_for_seconds = $divisor_for_minutes % 60;
	    $seconds = ceil($divisor_for_seconds);
	 
	    $obj = array(
	        "h" => (int) $hours,
	        "m" => (int) $minutes,
	        "s" => (int) $seconds,
	    );
	    return $obj;
	}
}
```

**Comment by GrahamCampbell** [comment]
I would encourage people to switch to Laravel's native rate limiting stuff if they want more features.

**Pull request closed without merging**